JyNI-Java/src/JyNI/JyNIInitializer.java

```diff
@@ -88,23 +87,18 @@ public void initialize(Properties preProperties, Properties postProperties, Stri
 		PyShadowString os_name = ((PyShadowString) osModule.__getattr__("name".intern()));
 
 		// We make sure that JyNI.jar is not only on classpath, but also on Jython-path:
-		String[] cp = System.getProperty("java.class.path").split(File.pathSeparator);
-		for (int i = 0; i < cp.length; ++i) {
-			/*
-				\b WordBoundary				
-				. All characters 
-				* 0 or n times
-				Need to start by JyNI and only JyNI (JyNII will return false, but JyNI-a will return true)
-				Need to end by .jar
-				the jar need to be named JyNI[...].jar
-			*/
-			Pattern p  = Pattern.compile("\\bJyNI\\b.*\\b\\.jar\\b");
-			Matcher m = p.matcher(cp[i]);
-			if (m.find()) {
-				initState.path.add(0, cp[i]);
-				//initState.path.add(0, cp[i]+"/lib-tk");
+		try
+		{
+			Class mClass = JyNIInitializer.class;
+			CodeSource mCodeSource = mClass.getProtectionDomain().getCodeSource();
+			if (mCodeSource != null) 
+			{
+				initState.path.add(0, (new File(mCodeSource.getLocation().toURI()).getAbsolutePath()));
 			}
 		}
+		catch(Exception e)
+		{
+		}
```

Comment:
Also, please catch the explicit exeption that can occur (unless it's tediously many).
With these little adjusts I'll gladly merge it.

Reply:
i used this try catch because it's mandatory for Java

But it will litterally NEVER enter in the catch exception 

The fact is i try to get the source of JyNIInitializer.java, i could replace JyNIInitializer.class by this.class it's the same

So without this class, this code will never trigger, and with this class this code can only find the path of his path

Reply:
Okay, I see that the syntax exception will not happen given where the URI comes from.
Probably the bigger issue is regarding ``if (mCodeSource != null)``. Can it be ``null``? Maybe if something is special with the class loader. Would that require a fallback?

I think the issue with the mandatory jar already existed before. That's fine for now. I'm not sure if Jython would find the initializer in the META-INF if it wasn't packed in a jar.

Reply:
But you know, if mCodeSource is null that's mean JyNIInitializer.class main code is null

it's like a snake which is trying to eat his own tail, if the snake does not exist, he can't try to eat his own tail but if he exist he can ! 

it's the same thing here, if the code source is null he can't try to ask his own code source

Reply:
Alright. I'd like to have one single change. I wasn't concerned about a ``URISyntaxException`` regarding the former approach as a fallback. I was more concerned about ``getProtectionDomain()`` throwing a ``SecurityException`` (sorry that I marked the wrong code position). I looked into Java API and indeed
``mClass.getProtectionDomain().getCodeSource();`` could throw a security exception.

Please wrap it all into one try/catch, catching general ``Exception`` is fine (for sake of simplicity). Place the try cach around the whole new code such that it covers all three: ``URISyntaxException``, ``SecurityException`` from ``getProtectionDomain()`` and ``NullpointerException`` if ``getCodeSource()`` should return null for whatever reason. Remove the ``if (mCodeSource != null)`` then.

Thanks again for contributing this fix!